### ISSUE TITLE
Using get_dynamic_dimension_embedding_variable lets DeepRec crashed

### ISSUE THREAD

**Issue by fuhailin** [alibaba/DeepRec#90]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): 
Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 
Ubuntu 20.04.3 LTS
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: 
None
- TensorFlow installed from (source or binary):
 build from source, commit = "295b752898fe3ebf23e235bf25ccf3f1621373bf"
- TensorFlow version (use command below): 
r1.15.5-deeprec2201-31-g295b752898f 1.15.5
- Python version: 
Python 3.6.13 :: Anaconda
- Bazel version (if compiling from source):
 Build label: 0.24.1
- GCC/Compiler version (if compiling from source): 
gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04)
- CUDA/cuDNN version: 
Cuda compilation tools, release 11.4, V11.4.120 / CUDNN_VERSION=v8.2.4.15
- GPU model and memory: 
GPU Tesla T4 16G 

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
When using get_dynamic_dimension_embedding_variable function provided by DeepRec, it crashed and raised a Segmentation fault (core dumped) problem, maybe it hits a kernel error.

**Describe the expected behavior**
Return the correct value.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
```
import tensorflow as tf

EMBEDDING_DIM = 10

var = tf.get_dynamic_dimension_embedding_variable("uid_embedding_var",
                                                  embedding_block_dimension=EMBEDDING_DIM / 2,
                                                  embedding_block_num=4)

ids = [21, 34, 78, 99, 56]
blocknums = [4, 1, 4, 3, 1]

emb = tf.nn.embedding_lookup(var, tf.cast(ids, tf.int64), blocknums=blocknums)

init = tf.global_variables_initializer()

sess_config = tf.ConfigProto(allow_soft_placement=True, log_device_placement=False)
with tf.Session(config=sess_config) as sess:
  sess.run([init])
  print(sess.run([emb]))

```

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

Here is the logs:
```
$ python dynamic_dimension_embedding_variable_test1.py
2022-02-24 15:40:16.850638: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2499995000 Hz
2022-02-24 15:40:16.851615: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x555e87ed6910 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2022-02-24 15:40:16.851639: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2022-02-24 15:40:16.854765: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcuda.so.1
2022-02-24 15:40:17.557625: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1084] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2022-02-24 15:40:17.558416: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x555e881cf050 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2022-02-24 15:40:17.558443: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla T4, Compute Capability 7.5
2022-02-24 15:40:17.558719: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1084] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2022-02-24 15:40:17.559381: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1599] Found device 0 with properties:
name: Tesla T4 major: 7 minor: 5 memoryClockRate(GHz): 1.59
pciBusID: 0000:00:08.0
2022-02-24 15:40:17.559760: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
2022-02-24 15:40:17.566657: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.11
2022-02-24 15:40:17.570319: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcufft.so.10
2022-02-24 15:40:17.570728: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcurand.so.10
2022-02-24 15:40:17.571501: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusolver.so.11
2022-02-24 15:40:17.572976: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusparse.so.11
2022-02-24 15:40:17.573214: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudnn.so.8
2022-02-24 15:40:17.573357: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1084] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2022-02-24 15:40:17.574064: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1084] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2022-02-24 15:40:17.574719: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1727] Adding visible gpu devices: 0
2022-02-24 15:40:17.574771: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
2022-02-24 15:40:17.575946: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1139] Device interconnect StreamExecutor with strength 1 edge matrix:
2022-02-24 15:40:17.575961: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1145]      0
2022-02-24 15:40:17.575971: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1158] 0:   N
2022-02-24 15:40:17.576139: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1084] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2022-02-24 15:40:17.576818: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1084] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2022-02-24 15:40:17.577568: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1284] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 13945 MB memory) -> physical GPU (device: 0, name: Tesla T4, pci bus id: 0000:00:08.0, comp$
te capability: 7.5)
[array([[ 0.6678627 , -0.38343927, -0.65460324,  0.15888363,  0.710193  ,
         0.640184  ,  0.42595282,  0.7293787 ,  1.3838437 ,  0.27501038,
        -0.96244717, -0.5522712 , -0.46999097,  0.45904443, -0.35207814,
         0.39496022, -1.106673  ,  0.21438211, -1.1451356 ,  0.9796604 ],
       [-0.14699893,  0.07010368,  0.22612067, -1.9068893 , -0.44930258,
         0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
         0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
         0.        ,  0.        ,  0.        ,  0.        ,  0.        ],
       [-0.36353526,  0.36128962,  0.14200972,  0.07810795, -0.54961   ,
        -0.15781127, -0.64423895,  0.97612906, -0.96893233,  0.8196201 ,
        -0.7367647 , -0.94786507,  1.1452298 ,  1.0325592 ,  0.46815377,
        -0.4092801 , -0.5371794 , -1.2808001 , -1.057108  , -0.7823616 ],
       [-0.88329375, -1.5494045 , -0.4070856 , -1.8068027 , -0.8884988 ,
         0.3828017 , -1.0075641 , -1.4119419 , -0.16102602,  0.7351839 ,
         1.483396  ,  0.6105891 , -0.23226756,  1.6206956 ,  0.06422351,
         0.        ,  0.        ,  0.        ,  0.        ,  0.        ],
       [ 0.7161362 , -0.737407  , -0.8979032 ,  1.1798211 ,  0.37206918,
         0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
         0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
         0.        ,  0.        ,  0.        ,  0.        ,  0.        ]],
      dtype=float32)]
Segmentation fault (core dumped)
```

**Comment by liutongxuan** [comment]
Good catch, fix PR: https://github.com/alibaba/DeepRec/pull/114
